app/models/odk/response_parser.rb

```diff
@@ -144,17 +144,41 @@ def add_group(xml_node, form_item, parent)
     end
 
     def add_answer(content, form_item, parent)
+      reject_invalid_options(content, form_item)
```

Comment:
Curious why you are doing this here instead of inside AnswerParser, which already has code for converting codes to nodes. It currently does go straight to Option ID though, so you'd have to edit it a bit.

See the line that says:

```
OptionNode.id_to_option_id(id)
```

You could change this to `OptionNode.where(option_set_id: form_item.option_set_id).id_to_option_id(id)` or something. You'd have to change the CodeMapper a bit to accept some scope information, but that seems. Reasonable. If this is confusing, let's pair on it.

I think we can also now get rid of the ` || OptionNode.old_id_to_option_id(id)` part and associated methods.

Reply:
Notes from pairing:
- this can be better addressed by updating answer.rb in a similar way
- old_id stuff will be done in a separate PR